### ISSUE TITLE
Weights file size not reducing!

### ISSUE THREAD

**Issue by sicarioakki** [diamour/darknet_project_privat#2]
By using this repo, there is no reduction in weights file size. And on  GPU, pruning is not happening at all
Could you please suggest something? @diamour

**Comment by Hao-Gong** [comment]
It works, you can check it on my README file. I updated it

**Comment by Hao-Gong** [comment]
Each con layer you can decide how to prune it in cfg file.

**Comment by sicarioakki** [comment]
yes, i am using the same cfg file used here. I see that you have used params like PRUNE, min and max. 
Are you telling that this pruning works only on a trained model?

**Comment by Hao-Gong** [comment]
Functionally it can work on trained or not trained model. But on a trained model, the not important weights can be pruned. Oh, i forget to say that the pruned weights only run on CPU not GPU, sorry. The project is designed for the CPU use case. The pruned model can save 50% time on CPU.

**Comment by sicarioakki** [comment]
Okay man! Thanks for your reply! I will try once more and let you know

**Comment by Hao-Gong** [comment]
Prune it on GPU. and test it on CPU.

**Comment by sicarioakki** [comment]
Okay will do that!

**Comment by sicarioakki** [comment]
Can the same pruning functionality be added for GPU?

**Comment by sicarioakki** [comment]
Also, can you tell me how much was the weights file reduction?

**Comment by abhigoku10** [comment]
@sicarioakki @diamour  hi even i am facing the same issue . I am able to run this source code on the CPU directly , i am able to get the reduction in processing time but the wight file is not reduced.Should i apply huffman coding to see the reducing in file size 

@diamour  is it only designed to cpu , if i have to make it work on gpu what all changes i have to make can you share the pointers 

@diamour thanks for the wonderful work on pruning the model
